containers-toolkit/Private/CommonToolUtilities.psm1

```diff
@@ -67,59 +100,303 @@ function Test-EmptyDirectory($path) {
     return ($itemCount.Count -eq 0)
 }
 
+function Get-ReleaseAssets {
+    [OutputType([PSCustomObject])]
+    param (
+        [string]$repo, # containers repo-owner/$repo-name
+        [string]$version,
+        [string]$OSArch
+    )
+
+    function Invoke-GitHubApi {
+        param($uri)
+        try {
+            Write-Debug "Invoking GitHub API. URI: $uri"
+            $response = Invoke-RestMethod -Uri "$uri" -Headers @{ "User-Agent" = "PowerShell" }
+            return $response
+        }
+        catch {
+            Throw "GitHub API error. URL: `"$uri`". Error: $($_.Exception.Message)"
+        }
+    }
+
+    Write-Debug "Getting release assets:`n`trepo: $repo`n`trelease version: $version`n`trelease architecture: $OSArch "
+    $baseApiUrl = "https://api.github.com/repos/$repo"
+    if ($version -eq "latest") {
+        $apiUrl = "$baseApiUrl/releases/latest"
+    }
+    else {
+        # We use this method to get the release assets for a specific version
+        # because creating a string for the version tag is not always consistent
+        # e.g. "v1.0.0" vs "1.0.0"
+        # The q parameter used for searching is not available in the GitHub API's /tags endpoint.
+        # GitHub's API for listing tags (/repos/{owner}/{repo}/tags) does not support querying or filtering
+        # directly through a q parameter like some other endpoints might (e.g., the search API).
+
+        # Get all releases tags
+        $response = Invoke-GitHubApi -Uri "$baseApiUrl/tags"
+        $releaseTag = $response | Where-Object { ($_.name.TrimStart("v")) -eq ($version.TrimStart("v")) }
+
+        if (-not $releaseTag) {
+            Throw "Couldn't find release tags for the provided version: '$version'"
+        }
+
+        if ($releaseTag.Count -gt 1) {
+            Write-Warning "Found multiple release tags for the provided version: '$version'. Using the first tag."
+        }
+
+        $releaseTagName = $releaseTag | Select-Object -First 1 -ExpandProperty name
+
+        # Get the release with the specified tag
+        Write-Debug "Release tag: $releaseTagName"
+        $apiUrl = "$baseApiUrl/releases/tags/$releaseTagName"
+    }
+    $response = Invoke-GitHubApi -Uri "$apiUrl"
+
+    # Filter list of assets by architecture and file name
+    $releaseAssets = $response | Select-Object -Property name, url, created_at, published_at, `
```

Comment:
nit: can fix the indentation for this section for readability?

```powershell
$releaseAssets = $response | Select-Object ... `
    ... `
    .... `
    .... = {
        $_.assets | Where-Object {
            (
                ($_.name -match .... ) -or
                (& ( .... ( ... .)))
            )
        } | ForEach-Object {
            ....
            ....
        }
    }
}
```

Reply:
PowerShell linter formats it this way